pkg/deployer/lib/controller.go

```diff
@@ -202,34 +203,27 @@ func (c *controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, nil
 	}
 
-	if di.Status.DeployItemPhase == lsv1alpha1.DeployItemPhaseSucceeded ||
-		di.Status.DeployItemPhase == lsv1alpha1.DeployItemPhaseFailed ||
-		di.Status.DeployItemPhase == "" {
-
+	if di.Status.Phase.IsFinal() || di.Status.Phase.IsEmpty() {
 		// The deployitem has a new jobID, but the phase is still finished from before
-
 		if di.DeletionTimestamp.IsZero() {
 			if di.Spec.UpdateOnChangeOnly &&
 				di.GetGeneration() == di.Status.ObservedGeneration &&
-				di.Status.DeployItemPhase == lsv1alpha1.DeployItemPhaseSucceeded {
+				di.Status.Phase == lsv1alpha1.DeployItemPhases.Succeeded &&
+				!testReconcileAnnotation {
```

Comment:
Why do we need this check? This must be always the case because otherwise the annotation is removed and the reconcile returns in line 203

Reply:
Hm, you are right. I missed that. But why did this change solve the mentioned problem? I'm confused.

Anyway, I will start splitting this PR into multiple smaller PRs and then refactor this again.

Reply:
Solve which mentioned problem?

Reply:
If you mean the conflict between test-reconcile and updateOnChangeOnly, I think this is not critical at the moment.